presto-parquet/src/main/java/com/facebook/presto/parquet/reader/ParquetReader.java

```diff
@@ -206,7 +209,18 @@ private ColumnChunk readPrimitive(PrimitiveField field)
             ParquetColumnChunk columnChunk = new ParquetColumnChunk(descriptor, buffer, 0);
             columnReader.setPageReader(columnChunk.readAllPages());
         }
-        return columnReader.readPrimitive(field);
+        try {
+            return columnReader.readPrimitive(field);
+        }
+        catch (UnsupportedOperationException e) {
+            throw new ParquetCorruptionException(format(
```

Comment:
This is not really a corruption issue. Sometimes users just alter their tables and hit this issue as well.